Sources/NIO/EventLoop.swift

```diff
@@ -914,6 +930,17 @@ public final class MultiThreadedEventLoopGroup: EventLoopGroup {
             }
         }
     }
+
+    public static func makeEventLoopFromCallingThread(_ callback: @escaping (EventLoop) -> Void) {
```

Comment:
Can we document acceptable uses of this and explain what will happen? I think it's especially important to explain what the exit conditions are (i.e. that this call will block the thread until the event loop is shut down).

Reply:
whoops, forgot the docs.

Reply:
It also feels weird calling this from MTELG. I wonder if making it internal here and surfacing it via some other namespace would be a better way to go: `<Something>.withCurrentThreadAsEventLoop { ... }`

Reply:
How about naming it something like ˋmakeSingleThreadedGroupFromCallingThreadˋ?

Reply:
> I felt it's not really worth adding a new top-level namespace for a function that'll be as rarely used as this one so I put it into MTELG and thought we can always (deprecate) move it elsewhere in the future. But 100% open to moving it.

Yeah I think that's fair: I'm fine with leaving it where it is. Although I do think `with` is a better prefix than `make` here. Make – at least in my mind – suggests returning something to the caller, with suggests it's providing an object scoped to the callback.

Reply:
@glbrntt yes, `with` over `make` is right! `make` is a factory.

@Lukasa happy with leaving it in MTELG?

Reply:
Yup.

Reply:
@glbrntt done & thanks Cory